ez/ez.go

```diff
@@ -98,9 +106,16 @@ func ConfigFileEnvFlag(ctx context.Context, cfg ConfigWithConfigPath, df Decoder
 		o(option)
 	}
 
-	fset, flagErr := flag.NewCmdLineSet(option.flagConfig, cfg)
-	if flagErr != nil {
-		return nil, fmt.Errorf("failed to register commandline flags: %s", flagErr)
+	var flagSrc dials.Source
+	if option.flagSubstitute != nil {
+		flagSrc = option.flagSubstitute
+	} else {
```

Comment:
Let's initialize `flagSrc` directly with `option.flagSubstitute`, the current `else` block just becomes `if flagSrc == nil {` ...

Reply:
Oh I like this a lot better. Thank you!